x-pack/plugins/siem/public/components/page/network/ip_overview/index.tsx

```diff
@@ -47,42 +46,34 @@ interface DescriptionList {
 }
 
 interface OwnProps {
-  ip: string;
   data: IpOverviewData;
+  flowTarget: FlowTarget;
+  ip: string;
   loading: boolean;
   type: networkModel.NetworkType;
+  updateFlowTargetAction: ActionCreator<{ flowTarget: FlowTarget }>;
 }
 
-interface IpOverviewReduxProps {
-  flowType: IpOverviewType;
-}
-
-interface IpOverViewDispatchProps {
-  updateIpOverviewFlowType: ActionCreator<{
-    flowType: IpOverviewType;
-  }>;
-}
-
-type IpOverviewProps = OwnProps & IpOverviewReduxProps & IpOverViewDispatchProps;
+type IpOverviewProps = OwnProps;
 
-class IpOverviewComponent extends React.PureComponent<IpOverviewProps> {
+export class IpOverview extends React.PureComponent<IpOverviewProps> {
   public render() {
-    const { ip, data, loading, flowType } = this.props;
-    const typeData: Overview = data[flowType]!;
+    const { ip, data, loading, flowTarget, updateFlowTargetAction } = this.props;
+    const typeData: Overview = data[flowTarget]!;
```

Comment:
Can you remove the `!` at this point or is that still needed? I wonder if that is covering up a bug or not

Reply:
Looks like this is still needed as `IP Overview` only support two of the four `FlowTarget`'s. I just tried updating the types of `IpOverviewData`, but it looks like there would be a bit more refactoring of that component to support removing the `!`.